SPECS/tdnf/tdnf-increase-timeout.patch

```diff
@@ -0,0 +1,92 @@
+From 24163dbd2edb027112ad57ba2ffe75365dc06a1d Mon Sep 17 00:00:00 2001
+From: Sam Meluch <sammeluch@microsoft.com>
+Date: Mon, 8 May 2023 14:57:38 -0700
+Subject: [PATCH] Add retry logic for connection failure
+
+---
+ client/remoterepo.c | 64 ++++++++++++++++++++++++++++++---------------
+ 1 file changed, 43 insertions(+), 21 deletions(-)
+
+diff --git a/client/remoterepo.c b/client/remoterepo.c
+index de0e04c..88e3f92 100644
+--- a/client/remoterepo.c
++++ b/client/remoterepo.c
+@@ -173,32 +173,54 @@ TDNFDownloadFile(
+         }
+     }
+ 
+-    fp = fopen(pszFile, "wb");
+-    if(!fp)
++    /* Double the connection timeout */
++    dwError = curl_easy_setopt(pCurl, CURLOPT_CONNECTTIMEOUT, 600);
++    BAIL_ON_TDNF_CURL_ERROR(dwError);
++
++    /* Add retry logic for connection failure*/
++    int retries = 2;
++    for( int i = 0; i < retries; i++)
+     {
+-        dwError = errno;
+-        BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+-    }
++        if (i > 0)
++        {
++            printf("Retrying from connection timeout failure\n");
++        }
++        fp = fopen(pszFile, "wb");
++        if(!fp)
++        {
++            dwError = errno;
++            BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
++        }
+ 
+-    dwError = curl_easy_setopt(pCurl, CURLOPT_WRITEDATA, fp);
+-    BAIL_ON_TDNF_CURL_ERROR(dwError);
++        dwError = curl_easy_setopt(pCurl, CURLOPT_WRITEDATA, fp);
++        BAIL_ON_TDNF_CURL_ERROR(dwError);
+ 
+-    dwError = curl_easy_perform(pCurl);
+-    BAIL_ON_TDNF_CURL_ERROR(dwError);
++        dwError = curl_easy_perform(pCurl);
++        if (i < (retries - 1) && dwError == 22)
```

Comment:
Is there a constant for "22" that you can use?

Reply:
Updated to use libcurl error code instead of "magic number"